json/src/lib.rs

```diff
@@ -123,6 +123,37 @@ pub struct LoadWalletResult {
     pub warning: Option<String>,
 }
 
+#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
+pub struct GetWalletInfoResult {
+    #[serde(rename = "walletname")]
+    pub wallet_name: String,
+    #[serde(rename = "walletversion")]
+    pub wallet_version: u32,
+    #[serde(with = "bitcoin::util::amount::serde::as_btc")]
+    pub balance: Amount,
+    #[serde(with = "bitcoin::util::amount::serde::as_btc")]
+    pub unconfirmed_balance: Amount,
+    #[serde(with = "bitcoin::util::amount::serde::as_btc")]
+    pub immature_balance: Amount,
+    #[serde(rename = "txcount")]
+    pub tx_count: usize,
+    #[serde(rename = "keypoololdest")]
+    pub keypool_oldest: usize,
+    #[serde(rename = "keypoolsize")]
+    pub keypool_size: usize,
+    #[serde(rename = "keypoolsize_hd_internal")]
+    pub keypool_size_hd_internal: usize,
+    pub unlocked_until: Option<usize>,
+    #[serde(rename = "paytxfee", with = "bitcoin::util::amount::serde::as_btc")]
+    pub pay_tx_fee: Amount,
+    #[serde(rename = "hdseedid")]
+    pub hd_seed_id: Option<String>,
+    #[serde(rename = "hdmasterkeyid")]
+    pub hd_master_key_id: Option<String>,
+    pub private_keys_enabled: bool,
+    pub avoid_reuse: Option<bool>,
```

Comment:
Does `hd_master_key_id` exist? I don't see it in docs or source ...

Seems like `avoid_reuse` can just be a `bool`, right? 

This implementation is also missing the `scanning` field:

```
  "scanning" : {                          (json object) current scanning details, or false if no scan is in progress
    "duration" : n,                       (numeric) elapsed seconds since scan start
    "progress" : n                        (numeric) scanning progress percentage [0.0, 1.0]
  }
```

But I'm not sure how to implement that because it could be either a bool or a struct. Similar scenario to https://github.com/rust-bitcoin/rust-bitcoincore-rpc/issues/124 where you have mixed content. We could always add this later, too.

Reply:
Those are possible to implement, but not easy. I don't know if serde does it automatically if you create an enum. I don't think it does because there is an implied false. I can try do a custom serde implementation after this is merged if you want. (I'll create a ticket for it when we merge this.)

Reply:
The `hdmasterkeyid` field seems to have been removed in 0.18. Since we only support 0.18 and 0.19, we shouldn't have it.

Reply:
> Does hd_master_key_id exist? I don't see it in docs or source ...

Thanks, as @stevenroose pointed out it was removed so I deleted it.

> Seems like avoid_reuse can just be a bool, right?

It didn't exist in v0.18 which is why I put it as an `Option<>`, but if there is a better way of handling this let me know (IIRC CI fails for v0.18 if it's not an option).

> This implementation is also missing the scanning field:

I was curious so I gave it a try and it seems to work. It cannot be properly tested through CI though because rescanning in regtest is too fast. I gave it a try on testnet and it seems to work as expected. I put it as a separate commit for now, so if you're not happy with the implementation I'll just drop it and someone else can take care of it later. If you're happy with it I can squash it. One downside with this implementation is that it's not really explicit about the fact that the boolean part of the enum can only be `false`, not sure if there is a better way.